app/webhook.go

```diff
@@ -323,6 +340,21 @@ func (a *App) CreateIncomingWebhookForChannel(creatorId string, channel *model.C
 		return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_username.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	if len(hook.WhiteIpList) > 0 {
+		if !(model.IsValidIpList(hook.WhiteIpList)) {
+			return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_white_ip_list.app_error", nil, "", http.StatusBadRequest)
+		}
+	} else if len(hook.HmacAlgorithm) > 0 {
+		if hook.IsValidModels() {
```

Comment:
nit: let's invert, and return an error for clarity

Reply:
Done in this commit.